backend/src/routes/classroom.ts

```diff
@@ -1,12 +1,56 @@
-import { Request, Response, Router } from 'express';
+import { NextFunction, Request, Response, Router } from 'express';
 import Classroom from 'models/classroom';
+import { makeError } from 'helpers/error';
+import { NOT_FOUND } from 'http-status';
 
 const router = Router();
 
-router.get('/', async (req: Request, res: Response) => {
-  const classrooms = await Classroom.findAll();
+router.get('/', (req: Request, res: Response, next: NextFunction) => {
+  Classroom.findAll()
+    .then(classrooms => res.json(classrooms))
+    .catch(error => next(error));
+});
+
+router.get('/:id', (req: Request, res: Response, next: NextFunction) => {
+  Classroom.findByPk(req.params.id)
+    .then(classroom => {
+      if (classroom) return res.json(classroom);
+      return next(makeError('Classroom with that id not found.', NOT_FOUND));
+    })
+    .catch(error => next(error));
+});
+
+router.post('/', (req, res: Response, next: NextFunction) => {
+  Classroom.create({
+    name: req.body.name,
+    capacity: req.body.capacity,
+  })
+    .then(classroom => res.json(classroom))
+    .catch(error => next(error));
+});
+
+router.put('/:id', (req: Request, res: Response, next: NextFunction) => {
+  Classroom.update(
+    {
+      name: req.body.name,
+      capacity: req.body.capacity,
+    },
+    { where: { id: req.params.id } },
+  )
+    .then(async results => {
+      if (results[0]) return res.json(await Classroom.findByPk(req.params.id));
```

Comment:
More a hint than anything else but the [update](https://sequelize.org/api/v6/class/src/model.js~model#static-method-update) method on a Sequelize model has a returning option that should return the updated models so there would be no need to do fetch it here.

Reply:
Thanks for that info, didn't see that. Changed it ❤️